sdk/cosmos/azure-cosmos/test/test_user_configs.py

```diff
@@ -68,6 +68,13 @@ def test_enable_endpoint_discovery(self):
         self.assertTrue(client_default.client_connection.connection_policy.EnableEndpointDiscovery)
         self.assertTrue(client_true.client_connection.connection_policy.EnableEndpointDiscovery)
 
+    def test_authentication_error(self):
+        try:
+            cosmos_client.CosmosClient(url=_test_config.host, credential="wrong_key")
+        except exceptions.CosmosHttpResponseError as e:
+            assert e.status_code == 401
+            assert e.reason == "Unauthorized"
```

Comment:
Do we not actually raise `azure.core.exceptions.AuthenticationError`?

Reply:
Nope, we raise the error in `_synchronized_request.py` as such:
![image](https://user-images.githubusercontent.com/30335873/226977699-bdc35956-62df-47c0-b1b9-1b43dab5fce1.png)